data_store/messaging/tf_messaging.py

```diff
@@ -305,8 +305,11 @@ def _wrong_type_failure_message(self, validation_failure: WrongTypeFailure) -> M
                 "Financial Year 2022/21 - Financial Year 2029/30",
                 ("Outcome Indicators (excluding " "footfall) and Footfall Indicator"),
             )
-            cell_index = self._get_cell_indexes_for_outcomes(validation_failure.failed_row)
-
+            cell_index = (
+                ""
+                if validation_failure.failed_row is None
+                else self._get_cell_indexes_for_outcomes(validation_failure.failed_row)
+            )
```

Comment:
`validation_failure.failed_row` can be `pb.Series | None`
`_get_cell_indexes_for_outcomes` accepts only `failed_row: pb.Series`

I think `validation_failure.failed_row` should never be `None` inside that `if` (when `sheet == Outcome`).
Not sure thought if that's a rule we can rely on.

I added tests to cover this scenario. Message() errors our when cell_index is "". 
I am also considered changing `if sheet == Outcome` to `if sheet == Outcome and validation_failure.failed_row is not None`. Perhaps it's the better approach.